src/NuGet.Protocol.Core.v3/HttpHandlerResourceV3Provider.cs

```diff
@@ -121,6 +130,33 @@ private class CredentialPromptWebRequestHandler : WebRequestHandler
                 }
             }
 
+            /// <summary>
+            /// Adds an explicit Authorization header to the request, if basic auth credentials for the package source
+            /// are available, no authorization header already exists on the request, and the request is for the
+            /// same scheme and authority as the service index.
+            /// </summary>
+            /// <remarks>
+            /// Resolves https://github.com/NuGet/Home/issues/742: NuGet should set PreAuthenticate to true to reduce the number of web requests
+            /// Note that PreAuthenticate doesn't do what you'd expect (it still doesn't use creds on the very first request),
+            /// so we set Authorization manually.
+            /// </remarks>
+            private void AddBasicAuthorizationHeaderIfAvailable(HttpRequestMessage request)
+            {
+                if (this.Credentials != null &&
```

Comment:
Please read and follow the coding standards for this repo

The short list - don't use this, use var, break long lines (if github shows the line to be longer than 120 chars break it), use using statements instead of specifically qualifying methods